service/src/main/java/tds/config/web/advice/ExceptionAdvice.java

```diff
@@ -0,0 +1,47 @@
+package tds.config.web.advice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import tds.config.web.resources.ExceptionMessageResource;
+
+import java.util.NoSuchElementException;
+
+@ControllerAdvice
+public class ExceptionAdvice {
+    private static final Logger LOG = LoggerFactory.getLogger(ExceptionAdvice.class);
+
+    // TODO:  Find better/more appropriate exception?  Make our own?
+    @ExceptionHandler(NoSuchElementException.class)
+    @ResponseStatus(value = HttpStatus.NOT_FOUND)
+    @ResponseBody
+    ResponseEntity<ExceptionMessageResource> handleNotFoundException(final NoSuchElementException ex) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Not Found exception occurred", ex);
+        } else {
+            LOG.error("Not Found exception occurred: {}", ex.getLocalizedMessage());
+        }
```

Comment:
I don't think we should be logging different things for error or debug.  The only time I use the isLevelEnabled is when there is additional logic in building up the log message that would be unnecessary. 

If we want to log this error, which I personally am not sure we do, then I suggest logging the full Exception or not logging the Exception at all.